Userland/Libraries/LibWeb/FileAPI/FileReader.idl

```diff
@@ -0,0 +1,35 @@
+#import <DOM/EventTarget.idl>
+#import <FileAPI/Blob.idl>
+
+// // https://w3c.github.io/FileAPI/#dfn-filereader
+[Exposed=(Window,Worker)]
+interface FileReader : EventTarget {
+  constructor();
+  // async read methods
+  undefined readAsArrayBuffer(Blob blob);
+  undefined readAsBinaryString(Blob blob);
+  undefined readAsText(Blob blob, optional DOMString encoding);
+  undefined readAsDataURL(Blob blob);
+
+  undefined abort();
+
+  // states
+  const unsigned short EMPTY = 0;
+  const unsigned short LOADING = 1;
+  const unsigned short DONE = 2;
+
+  readonly attribute unsigned short readyState;
+
+  // File or Blob data
+  readonly attribute (DOMString or ArrayBuffer)? result;
+
+  readonly attribute DOMException? error;
+
+  // event handler content attributes
+  attribute EventHandler onloadstart;
+  attribute EventHandler onprogress;
+  attribute EventHandler onload;
+  attribute EventHandler onabort;
+  attribute EventHandler onerror;
+  attribute EventHandler onloadend;
```

Comment:
Please re-indent this with 4 spaces. I know copy-pasting the spec sometimes gives you two, sometimes 4, sometimes who knows, but we consistently use 4. :^)

Reply:
Yeah, I actually used to do that since it seems more common, but I noticed doing `DeprecatedString` cleanup quite a few IDL files which don't do that, some of which have been added recently:
 * Libraries/LibWeb/Crypto/Crypto.idl
 * Libraries/LibWeb/HTML/TextMetrics.idl
 * Libraries/LibWeb/HTML/NavigatorOnLine.idl
 * Libraries/LibWeb/HTML/NavigatorConcurrentHardware.idl
 * Libraries/LibWeb/HTML/NavigationHistoryEntry.idl
 * Libraries/LibWeb/HTML/NavigationDestination.idl
 * Libraries/LibWeb/HTML/NavigationCurrentEntryChangeEvent.idl
 * Libraries/LibWeb/HTML/Navigation.idl
 
etc.  I figured I would just copy past verbatim. I can change this here, and go through all of the others in a separate MR?

Reply:
Ah, those should be 4 too but it's easy not to notice.

Reply:
Cool, I've amended the IDL file I'm adding, I'll make a follow up to go through and do the rest.